### PR TITLE
test[cartesian]: update hypothesis and fix the health check issues found in daily CI

### PR DESCRIPTION
**Pull request by DropD**
## Description

The daily CI [found](https://github.com/GridTools/gt4py/actions/runs/6092626057/job/16531045858#step:8:2297) that after updating to hypothesis 6.82.1 some tests in cartesian are failing hypothesis health checks. This PR tries to fix that.

## Requirements

- [X] All fixes and/or new features come with corresponding tests.
- [X] Important design decisions have been documented in the approriate ADR inside the [docs/development/ADRs/](docs/development/ADRs/Index.md) folder.

### CODE REVIEW

**Comment by DropD** [comment]
Seems that the new version caused the health check for multiple executers to trigger on parametrized tests. From the tests in hypothesis itself it looked like this might not happen if parametrization is the outer decorator (applied after @given).

**Review by egparedes — APPROVED** [review]
It looks good to me.

File: requirements-dev.in [hunk 1]
```diff
@@ -5,7 +5,7 @@
 ## for pkg in ['hypothesis', 'pytest']:
 ##    print(re.search(f"\n({pkg} *[=>~!].*)\n", versions)[1])
 ##]]]
-hypothesis>=6.0.0
```
Comment (egparedes) [inline requirements-dev.in:8]: Whenever any requirement is modified, the tox workflow updating all config files need to be executed as explained here: https://github.com/GridTools/gt4py/blob/main/docs/development/tools/requirements.md

**Review by egparedes — CHANGES_REQUESTED** [review]
We need to run the requirements update workflow before merging.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -146,7 +146,7 @@ repos:
   ## version = re.search('mypy==([0-9\.]*)', open("constraints.txt").read())[1] 
   ## print(f"#========= FROM constraints.txt: v{version} =========")
   ##]]]
-  #========= FROM constraints.txt: v1.5.0 =========
+  #========= FROM constraints.txt: v1.5.1 =========
   ##[[[end]]]
   rev: v1.5.0  # MUST match version ^^^^ in constraints.txt (if the mirror is up-to-date)
```
Comment (egparedes) [inline .pre-commit-config.yaml:151]:
```suggestion
  rev: v1.5.1  # MUST match version ^^^^ in constraints.txt (if the mirror is up-to-date)
```

**Review by egparedes — APPROVED** [review]
The mypy version in pre-commit-config needs to be updated  manually because the hook is defined in the mirror repository and it's not always up-to-date.

This has been now fixed so the PR is ready to be merged IMHO.

It looks good to me

**Pull request merged**